account_invoice_fiscal_position_update/models/account_invoice.py

```diff
@@ -27,28 +27,30 @@ def fiscal_position_change(self):
         lines_without_product = []
         fp = self.fiscal_position_id
         inv_type = self.type
+        # map also invoice account if set on fiscal position
+        if fp:
+            self.account_id = fp.map_account(self.account_id)
         for line in self.invoice_line_ids:
             if line.product_id:
+                account = line.get_invoice_line_account(
+                    inv_type, line.product_id, fp, self.company_id
+                )
+                # Use mapping of line account in case it was overwritten
+                # the product account
+                line_account = fp.map_account(line.account_id)
```

Comment:
@pedrobaeza I mean this one, in case we overwrite the invoice line account with a different account (not the one taken from product/category), the normal mapping is reverting it, mapping the account from product/category.

Reply:
No, that would be incorrect. Imagine you have product with account A. Previous fiscal position maps A to B, so B is the one before switching to another fiscal position. Imagine that one maps A to C, so as is, the final account is C, which is correct. With this code, it will try to map B, so leaving B, and that's incorrect.

Reply:
Revert it. Thanks